components/forms/formr-part-b/viewSections/View.tsx

```diff
@@ -80,7 +80,7 @@ const View = ({ canEdit, history }: IView) => {
         <ViewSection4 {...viewSectionProps} />
         <ViewSection5 {...viewSectionProps} />
         <ViewSection6 {...viewSectionProps} />
-        {isCovidDeclarationNull ? <ViewSection7 {...viewSectionProps} /> : null}
+        {!canEdit && !isCovidDeclarationNull ? <ViewSection7 {...viewSectionProps} /> : null}
```

Comment:
It might be worth making things a bit clearer here.. You could swap the logic round a bit so you are not having to negate a isNull variable . Something like renaming the `... isCovidDeclarationNull ... `to `const covidDeclarations = formData.haveCovidDeclarations` (on line 32). The Types for this variable will be `string, boolean, null `so this will still work to render/not render the covid section.
Then on line 83 you can do something like `{!canEdit && covidDeclarations && (<ViewSection7 {...viewSectionProps} />)}`